lib/plugin.lazy.js

```diff
@@ -0,0 +1,139 @@
+  window.addEventListener('error', SentryMock.captureException)
+
+  // Inject Sentry mock to the context as $sentry
+  inject('sentry', SentryMock)
+  <% } %>
+
+  const loadSentryHook = () => !loaded && loadSentry(ctx, inject)
+
+  <% if (options.lazy.injectLoadHook) { %>
+  inject('sentryLoad', loadSentryHook)
+  ctx.$sentryLoad = loadSentryHook
+  <% } else { %>
+  window.onNuxtReady(loadSentryHook)
+  <% } %>
+
+  const sentryReadyPromise = new Promise(resolve => {
+    sentryReadyResolve = resolve
+  })
+
+  const sentryReady = () => sentryReadyPromise
+
+  inject('sentryReady', sentryReady)
+  ctx.$sentryReady = sentryReady
+}
+
+async function loadSentry(ctx, inject) {
+  loaded = true
+
+  if (!window.<%= globals.nuxt %>) {
+    <% if (options.dev) { %>
+    console.warn(`$sentryLoad was called but window.<%= globals.nuxt %> is not available, delaying sentry loading until onNuxtReady callback. Do you really need to use lazy loading for Sentry?`)
+    <% } %>
+    window.<%= globals.readyCallback %>Cbs.push(() => loadSentry(ctx, inject))
+    return
+  }
+
+  <%
+  const magicComments = [`webpackChunkName: '${options.lazy.chunkName}'`]
+  if (options.lazy.webpackPrefetch) {
+    magicComments.push('webpackPrefetch: true')
+  }
+  if (options.lazy.webpackPreload) {
+    magicComments.push('webpackPreload: true')
+  }
+  %>
+  const Sentry = await import(/* <%= magicComments.join(', ') %> */ '@sentry/browser')
+  const { <%= Object.keys(options.integrations).join(', ') %> } = await import(/* <%= magicComments.join(', ') %> */ '@sentry/integrations')
+
+  <% if (options.initialize) { %>// Initialize sentry
+  const config = <%= serialize(options.config) %>
+  config.integrations = [
+    <%= Object.entries(options.integrations).map(([name, integration]) => {
+      if (name === 'Vue') {
+        return `new ${name}({Vue: VueLib, ...${serialize(integration)}})`
+      }
+
+      const integrationOptions = Object.entries(integration).map(([key, option]) => 
+        typeof option === 'function'
+        ? `${key}:${serializeFunction(option)}`
+        : `${key}:${serialize(option)}`
+      )
+
+      return `new ${name}({${integrationOptions.join(',')}})`
+    }).join(',\n      ')%>
+  ]
+
+  Sentry.init(config)
+  <% } %>
+
+  <% if (options.lazy.injectMock) { %>
+  window.removeEventListener('error', SentryMock.captureException)
+
+  delayedCalls.forEach(([methodName, args]) => Sentry[methodName].apply(Sentry, args))
+  <% } %>
+
+  injectOverwrite(ctx, 'sentry', Sentry)
+
+  sentryReadyResolve(Sentry)
+
+  // help gc
+  <% if (options.lazy.injectMock) { %>
+  // Dont unset delayedCalls & SentryMock during
+  // development, this will cause HMR issues
+  <% if (!options.dev) { %>
+  delayedCalls = undefined
+  SentryMock = undefined
+  <% } else { %>
+  delayedCalls = []
+  <% } %>
+  <% } %>
+  sentryReadyResolve = undefined
+}
+
+
+// Temporary inject into Nuxt without using Nuxt's inject method,
+// so we can later inject the final value
```

Comment:
This comment is outdated as now we are using this function to inject the final value.

But also, consider passing original `inject` to it, calling it and then just overwriting context[key] and $options as only those two really need custom handling.

Reply:
Addressed with 1eafbf2c0537640c98204654f8f1ebc23b0dde97